### PR TITLE
Including `examples` as subpackage in porepy using `setuptools`

### PR DESCRIPTION
**Pull request by vlipovac**
## Proposed changes

The `examples` package is now included in the set-up and is available via `from porepy import examples`.
Requires rebuilding the wheel (`pip install .` not `pip install -e .`)

## Types of changes

- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Testing (contribution related to testing of existing or new functionality).
- [ ] Documentation (contribution related to adding, improving, or fixing documentation).
- [ ] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [x] Other: Modification of set-up and installation

## Checklist

- [ ] The documentation is up-to-date.
- [ ] Static typing is included in the update.
- [X] This PR does not duplicate existing functionality.
- [ ] The update is covered by the test suite (including tests added in the PR).
- [ ] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

**Comment by vlipovac** [comment]
`examples/terzaghi_biot.py` and `examples/mandel_biot.py` changed because I ran `isort, black, flake8` in `examples`.

File: setup.py [hunk 1]
```diff
@@ -29,8 +41,8 @@ def read(fname):
             "applications/md_grids/gmsh_file_library/**/*.geo",
         ],
     },
-    packages=find_packages("src"),
-    package_dir={"": "src"},
+    packages=packages,
+    package_dir=package_dir,
     py_modules=[
```
[the remark this answers is not in the document]
Comment (IvarStefansson) [reply setup.py:46]: I agree, py_modules should be redundant.

**Comment by vlipovac** [comment]
Fixed the mypy issues related to our code.
Remaining mypy issues are about `pyplot.colors` and `pyplot.cm`, which I am not sure how to deal with.

**Comment by IvarStefansson** [comment]
> Fixed the mypy issues related to our code. Remaining mypy issues are about `pyplot.colors` and `pyplot.cm`, which I am not sure how to deal with.

Can we ignore them?

**Comment by IvarStefansson** [comment]
With the new structure, how would I import an example? Do we need/wish to add anything to any init files?

**Comment by vlipovac** [comment]
> With the new structure, how would I import an example? Do we need/wish to add anything to any init files?

@IvarStefansson `from porepy import examples` for a general import
For specific objects in the examples, `from porepy.examples.mandel_biot import *` works for example.

I would not import examples by default in `porepy/__init__.py`, s.t. `example` is a member of `pp` in the interpreter,
for the reasons I mentioned here https://github.com/pmgbergen/porepy/issues/1113#issuecomment-1911797141

**Comment by vlipovac** [comment]
> > Fixed the mypy issues related to our code. Remaining mypy issues are about `pyplot.colors` and `pyplot.cm`, which I am not sure how to deal with.
> 
> Can we ignore them?

@IvarStefansson I added `# type:ignore [index]` and `[attr-defined]` to where applicable

**Review by keileg — APPROVED** [review]
Looks good to me. I have also talked to @IvarStefansson, who agrees this is fine. 

Thanks a lot for the effort!

**Pull request merged**